File: core/src/main/java/co/infinum/collar/annotations/TrackAttribute.java
package co.infinum.collar.annotations;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

import co.infinum.collar.Trackable;

/**
 * Used along with {@link Trackable} interface
 */
@Retention(RetentionPolicy.RUNTIME)
@Target({ElementType.PARAMETER, ElementType.METHOD})
public @interface TrackAttribute {
}

File: annotations/src/main/kotlin/co/infinum/collar/annotations/TrackAttribute.kt
package co.infinum.collar.annotations

/**
 * Used along with Trackable interface
 */
@Retention(AnnotationRetention.RUNTIME)
@Target(AnnotationTarget.VALUE_PARAMETER, AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
annotation class TrackAttribute